WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs

```diff
@@ -161,7 +167,51 @@ private void Load(string filename, int id, int nameIndex, Dictionary<long, long>
             {
                 Load(filename, row => dictionary.Add(row.GetInt(id), row.GetInt(nameIndex)));
             }
-            
+
+            private void Load(string filename, string fieldName, Dictionary<long, string> dictionary)
+            {
+                var dbcd = new DBCD.DBCD(dbcProvider, dbdProvider);
```

Comment:
Can `dbcd` be constructed just once?

Also, instead of manually constructing `DBDProvider`, you can take advantage of the editor's Dependency Injection system:
1. Add `[AutoRegister]` attributes to `DBDProvider` and `DBCProvider` class definition
2. Add `DBCD.DBCD dbc` argument to `DbcStore` constructor

Done! This object shall be created automatically with resolved dependencies.